nav2_costmap_2d/src/costmap_2d.cpp

```diff
@@ -95,11 +95,17 @@ void Costmap2D::resetMaps()
 }
 
 void Costmap2D::resetMap(unsigned int x0, unsigned int y0, unsigned int xn, unsigned int yn)
+{
+  resetMapToValue(x0, y0, xn, yn, default_value_);
+}
+
+void Costmap2D::resetMapToValue(
+  unsigned int x0, unsigned int y0, unsigned int xn, unsigned int yn, unsigned char value)
 {
   std::unique_lock<mutex_t> lock(*(access_));
   unsigned int len = xn - x0;
   for (unsigned int y = y0 * size_x_ + x0; y < yn * size_x_ + x0; y += size_x_) {
-    memset(costmap_ + y, default_value_, len * sizeof(unsigned char));
+    memset(costmap_ + y, value, len * sizeof(unsigned char));
```

Comment:
I like this refactor, but I wonder if ROS1 was doing the wrong thing resetting the map to NO_INFORMATION. It kind of seems like setting it to the default_value might be more correct. Still, let's stick with ROS1 behavior unless we're fairly sure.

Reply:
I wouldn't do this - I'd make sure to use the default value that can be set from a parameterization. In not all situations do you want a clear to result in no information. There's a parameter around called track_unknown_space which sets the default value to FREE or UNKNOWN (which you call here as NO_INFORMATION). This would be relevent for robotics applications where you want the robot to only be able to plan and control in areas that are known to be free, rather than not knowing anything about it.

For instance, I'm a walking robot (or a warehouse robot) and I clear my costmaps, I dont want to back up into unknown space if I dont have sensors back there or else I could fall down a stairwell after my costmaps are cleared. On the counter, I'm a walking or warehouse robot in a particularly sensitive area of the warehouse where it might be dangerous to go somewhere or a hospital, then I want to only go places in my map (that has been created) that I know are free based on that mapping knowledge. To stop it from going elsewhere in unknown or unexplored space (like the ER), I would set track_unknown_space to false so that free in map is free in reality and then tell my planner to disallow the robot from planning into those zones that are unmapped intentionally to not allow the robot to go anywhere unmapped or unexplored (intentionally). 

tl;dr, I'd recommend making it `default_value_` otherwise, you should go through and just completely remove the track_unknown_space logic and parameters so people dont think that will still work

Reply:
Thank for pointing this out. I'll change it to use the `default_value_`.